laiaaa0/assignment4/Map.cpp

```diff
@@ -77,18 +61,22 @@ std::pair<int,int> Map::GetNextUnvisited(int x, int y, const std::vector<std::ve
             }
         }
     }
-    next_unvisited.first = this->num_rows_;
-    next_unvisited.second = this->num_columns_;
-
+    next_unvisited.first = this->map_tiles_.size();
+    if (next_unvisited.first < 1) next_unvisited.second = 0;
+    else {
+      next_unvisited.second = this->map_tiles_[0].size();
+    }
     return next_unvisited;
 }
 
 int Map::FindNumberIslands() {
-    std::vector<std::vector<bool> > visited(this->num_rows_, std::vector<bool> (this->num_columns_,false));
+    int num_columns = 0;
+    if (this->map_tiles_.size()>0) num_columns = this->map_tiles_[0].size();
+    std::vector<std::vector<bool> > visited(this->map_tiles_.size(), std::vector<bool> (num_columns,false));
     int x = 0, y = 0;
     int num_islands = 0;
-    while (x<this->num_rows_ && y < this->num_columns_){
-        if (this->ExpandNode (x,y,visited)) num_islands++;
+    while (x<this->map_tiles_.size() && y < this->map_tiles_[x].size()){
```

Comment:
Checking for y in range is enough, and saves you setting `next_unvisited.second` inside `Map::GetNextUnvisited()`